scipy/optimize/_qap.py

```diff
@@ -284,13 +290,19 @@ def _quadratic_assignment_faq(A, B,
         ``partial_match[i, 1]`` of `B`. The array has shape ``(m, 2)``, where
         ``m`` is not greater than the number of nodes, :math:`n`.
 
-    rng : {None, int, `numpy.random.Generator`, `numpy.random.RandomState`}, optional
-        If `seed` is None (or `np.random`), the `numpy.random.RandomState`
-        singleton is used.
-        If `seed` is an int, a new ``RandomState`` instance is used,
-        seeded with `seed`.
-        If `seed` is already a ``Generator`` or ``RandomState`` instance then
-        that instance is used.
+    rng : {None, int, `numpy.random.Generator`}, optional
+        Pseudorandom number generator state. When `rng` is None, a new
+        `numpy.random.Generator` is created using entropy from the
+        operating system. Types other than `numpy.random.Generator` are
+        passed to `numpy.random.default_rng` to instantiate a ``Generator``.
+
+        .. versionchanged:: 1.15.0
+            As part of the `SPEC-007 <https://scientific-python.org/specs/spec-0007/>`_
+            transition from use of `numpy.random.RandomState` to
+            `numpy.random.Generator` is occurring. For an interim period you can
+            continue to supply `np.random.RandomState` to this function. After this
+            period using `np.random.RandomState` will raise an exception.
+
```

Comment:
I this this is my fault, and I regret trying to copy-paste full documentation from public functions into private methods. Besides being superfluous, it bloats files making them a little hard to work with, and it's not a great use of time to try to keep in sync. So if we're going to adjust these, how about:
```suggestion
        Pseudorandom number generator state. See `quadratic_assignment` for details.
```